src/Microsoft.Health.Dicom.Functions/BlobMigration/CleanupDeletedDurableFunction.Activity.cs

```diff
@@ -0,0 +1,89 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Dicom.Core.Features.ChangeFeed;
+using Microsoft.Health.Dicom.Core.Features.Model;
+using Microsoft.Health.Dicom.Functions.BlobMigration.Models;
+
+namespace Microsoft.Health.Dicom.Functions.BlobMigration;
+
+public partial class CleanupDeletedDurableFunction
+{
+    /// <summary>
+    /// Asynchronously retrieves the next set of instance batches based on the configured options.
+    /// </summary>
+    /// <param name="arguments">The options for configuring the batches.</param>
+    /// <param name="logger">A diagnostic logger.</param>
+    /// <returns>
+    /// A task representing the asynchronous get operation. The value of its <see cref="Task{TResult}.Result"/>
+    /// </returns>
+    [FunctionName(nameof(GetDeletedChangeFeedInstanceBatchesAsync))]
+    public async Task<IReadOnlyCollection<ChangeFeedEntry>> GetDeletedChangeFeedInstanceBatchesAsync(
+        [ActivityTrigger] CleanupDeletedBatchArguments arguments,
+        ILogger logger)
+    {
+        EnsureArg.IsNotNull(arguments, nameof(arguments));
+        EnsureArg.IsNotNull(logger, nameof(logger));
+
+        return await _changeFeedStore.GetDeletedChangeFeedByWatermarkOrTimeStampAsync(
```

Comment:
Where is this? lol

Reply:
I didnt merge my sql PR, merged now. Its in main